### ISSUE TITLE
no-deprecated-deno-api doesn't catch deprecations in types

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#1014]
### Lint Name

no-deprecated-deno-api

### Code Snippet

```ts
    const file: Deno.File = await Deno.open(tempFile, {
      create: true,
      write: true,
      read: true,
    });
```

### Expected Result
Should warn about `Deno.File` being deprecated

### Actual Result

No warning

### Version

Deno 1.19.0
